python/plugins/processing/gui/HistoryDialog.py

```diff
@@ -179,45 +187,114 @@ def fillTree(self):
     def executeAlgorithm(self):
         item = self.tree.currentItem()
         if isinstance(item, TreeLogEntryItem):
-            if item.isAlg:
+            if item.as_python_command():
                 script = 'import processing\n'
                 # adding to this list? Also update the BatchPanel.py imports!!
                 script += 'from qgis.core import QgsProcessingOutputLayerDefinition, QgsProcessingFeatureSourceDefinition, QgsProperty, QgsCoordinateReferenceSystem, QgsFeatureRequest\n'
                 script += 'from qgis.PyQt.QtCore import QDate, QTime, QDateTime\n'
                 script += 'from qgis.PyQt.QtGui import QColor\n'
-                script += item.entry.text.replace('processing.run(', 'processing.execAlgorithmDialog(')
+                script += item.as_python_command().replace('processing.run(', 'processing.execAlgorithmDialog(')
                 self.close()
                 exec(script)
 
     def changeText(self):
         item = self.tree.currentItem()
         if isinstance(item, TreeLogEntryItem):
-            self.text.setText('"""\n' + self.tr('Double-click on the history item or paste the command below to re-run the algorithm') + '\n"""\n\n' +
-                              item.entry.text.replace(LOG_SEPARATOR, '\n'))
+            self.text.setText('"""\n' + self.tr(
+                'Double-click on the history item or paste the command below to re-run the algorithm') + '\n"""\n\n' +
+                item.as_python_command())
         else:
             self.text.setText('')
 
     def createTest(self):
         item = self.tree.currentItem()
         if isinstance(item, TreeLogEntryItem):
-            if item.isAlg:
-                TestTools.createTest(item.entry.text)
+            if item.as_python_command():
+                TestTools.createTest(item.as_python_command())
+
+    def copy_text(self, text: str):
```

Comment:
For consistency's sake, should this be renamed to copyText?

Reply:
I just can't bring myself to use that capitalization style for Python code :wink: